### PR TITLE
Update nextjs guide `global-error.tsx` example types

### PR DESCRIPTION
**Pull request by edgarlr**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

- Fix type error in Nextjs manual setup guide `global-error.tsx` example. Importing the Error component as `import Error from "next/error"`  interferes and overrides the global `Error ` type used in the props

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by shanamatthews** [comment]
Thanks for the fix @edgarlr !

@lforst - can you double check this is correct? Good to merge after you approve :)

**Review by shanamatthews — APPROVED** [review]

**Comment by lforst** [comment]
Thank you for your contribution! @edgarlr

**Pull request merged**
